src/pip/_internal/network/auth.py

```diff
@@ -110,7 +112,7 @@ def _get_index_url(self, url):
         return None
 
     def _get_new_credentials(self, original_url, allow_netrc=True,
-                             allow_keyring=True):
+                             allow_keyring=False):
```

Comment:
Have we looked at the impact this might have on keyring providers and associated package index assumptions? From #8103 it seems like Azure DevOps gives us a redirect, not a 401, when authentication is needed, and so may depend on our eager usage of `artifacts-keyring`. Not that it should necessarily block us, but this may not be a low impact change.

Reply:
Honestly, I have no idea. My drive is to fix this for users who use the default pypi.org index. While I've tried to make it work for the others as well, I don't have access to such setups.

Reply:
Great point. Can we just disable the initial keyring support for pypi-related URLs? We keep them [here](https://github.com/pypa/pip/blob/cdeb377a3633c9a79e4bd08f5e0026c8be7a7143/src/pip/_internal/models/index.py#L29-L34). I think that would narrow the scope of this enough that we can defer investigating my previous concern.